react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import Title from './title';
+import ImageShift from './image-shift';
+import ProgressSquares from './progress-squares';
+
+const interval = 1500;
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      intervalId: undefined,
```

Comment:
When setting an initial value it is usually better to use `null` over `undefined`. `undefined` can happen for a few different reasons vs `null` which can only happen if explicitly set. Much clearer intention with `null`